packages/server/src/orchestrator/atom.ts

```diff
@@ -1,106 +1,20 @@
-import * as R from 'ramda';
-
-import { EventEmitter } from 'events';
-import { Operation } from 'effection';
-import { Mailbox } from '@bigtest/effection';
-
-import { OrchestratorState } from './state';
-
-export class Atom {
-  private initial: OrchestratorState = {
-    manifest: {
-      description: 'None',
-      fileName: '<init>',
-      steps: [],
-      assertions: [],
-      children: []
-    },
-    agents: {},
-    testRuns: {},
-  };
-
-  private state = this.initial;
-
-  private subscriptions = new EventEmitter();
-
-  get(): OrchestratorState {
-    return this.state;
-  }
-
-  reset(initializer?: (initial: OrchestratorState, current: OrchestratorState) => OrchestratorState) {
-    if (!initializer) {
-      initializer = initial => initial;
-    }
-    this.state = initializer(this.initial, this.state);
-    this.subscriptions.removeAllListeners();
-  }
-
-  update(fn: (state: OrchestratorState) => OrchestratorState) {
-    this.state = fn(this.get());
-    this.subscriptions.emit('state', this.state);
-  }
-
-  slice<T>(path: string[]): Slice<T> {
-    return new Slice(this, path);
-  }
-
-  *each(fn: (state: OrchestratorState) => Operation) {
-    let mailbox = yield Mailbox.subscribe(this.subscriptions, "state");
-
-    while (true) {
-      let { args: [state] } = yield mailbox.receive();
-      yield fn(state);
-    }
+import { Atom as _Atom, Slice as _Slice } from "@bigtest/atom";
+import { OrchestratorState } from "./state";
+
+export class Atom extends _Atom<OrchestratorState> {
```

Comment:
Can we do this without extending atom? Maybe by having a function `OrchestratorAtom` that returns `Atom<OrchestratorState>`?

Reply:
@cowboyd like this?

```ts
export const OrchestratorAtom = () => new Atom({
  manifest: {
    description: "None",
    fileName: "<init>",
    steps: [],
    assertions: [],
    children: [],
  },
  agents: {},
  testRuns: {},
});
```

Reply:
Pretty much, although you might need to explicitly mark it so that TS can figure out that `S` === `OrchestratorState`

```ts
function OrchestratorAtom(): Atom<OrchestratorState> {
  return new Atom({
   manifest: {
    description: "None",
    fileName: "<init>",
    steps: [],
    assertions: [],
    children: [],
  },
  agents: {},
  testRuns: {},
  });
}
```

Reply:
@cowboyd done